File: client-clojure/src/sirqul_io_t_platform/specs/chart_data.clj
(ns sirqul-io-t-platform.specs.chart-data
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            )
  (:import (java.io File)))


(def chart-data-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :identifier) string?
   (ds/opt :idAttribute) string?
   (ds/opt :label) string?
   (ds/opt :maxValue) int?
   (ds/opt :totalCount) int?
   (ds/opt :items) (s/coll-of any?)
   (ds/opt :aggregatedItems) (s/coll-of any?)
   (ds/opt :caption) string?
   (ds/opt :showAnchor) string?
   (ds/opt :anchorAlpha) string?
   (ds/opt :getxAxisName) string?
   (ds/opt :getyAxisMinValue) string?
   (ds/opt :getyAxisName) string?
   (ds/opt :decimalPrecision) string?
   (ds/opt :formatNumberScale) string?
   (ds/opt :numberPrefix) string?
   (ds/opt :showNames) string?
   (ds/opt :showValues) string?
   (ds/opt :showAlternateHGridColor) string?
   (ds/opt :alternateHGridColor) string?
   (ds/opt :divLineColor) string?
   (ds/opt :divLineAlpha) string?
   (ds/opt :alternateHGridAlpha) string?
   (ds/opt :rotateNames) string?
   (ds/opt :returning) string?
   })

(def chart-data-spec
  (ds/spec
    {:name ::chart-data
     :spec chart-data-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/ChartData.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ChartData._

case class ChartData (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
identifier: Option[String],
idAttribute: Option[String],
label: Option[String],
maxValue: Option[Long],
totalCount: Option[Long],
items: Option[List[Any]],
aggregatedItems: Option[List[Any]],
caption: Option[String],
showAnchor: Option[String],
anchorAlpha: Option[String],
getxAxisName: Option[String],
getyAxisMinValue: Option[String],
getyAxisName: Option[String],
decimalPrecision: Option[String],
formatNumberScale: Option[String],
numberPrefix: Option[String],
showNames: Option[String],
showValues: Option[String],
showAlternateHGridColor: Option[String],
alternateHGridColor: Option[String],
divLineColor: Option[String],
divLineAlpha: Option[String],
alternateHGridAlpha: Option[String],
rotateNames: Option[String],
returning: Option[String])

object ChartData {
  import DateTimeCodecs._

  implicit val ChartDataCodecJson: CodecJson[ChartData] = CodecJson.derive[ChartData]
  implicit val ChartDataDecoder: EntityDecoder[ChartData] = jsonOf[ChartData]
  implicit val ChartDataEncoder: EntityEncoder[ChartData] = jsonEncoderOf[ChartData]
}
